src/tests/ftest/scripts/main.sh

```diff
@@ -237,6 +237,7 @@ fi
 
 # now run it!
 # shellcheck disable=SC2086
+env
```

Comment:
You need to export `WITH_VALGRIND` here if you want it in `launch.py`'s environment.
```suggestion
env
export WITH_VALGRIND
```